src/coreclr/src/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -0,0 +1,224 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Threading;
+using System.Runtime.CompilerServices;
+using Internal.Runtime.CompilerServices;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// Enumeration of flags for <see cref="ComWrappers.GetOrCreateComInterfaceForObject(object, CreateComInterfaceFlags)"/>.
+    /// </summary>
+    [Flags]
+    public enum CreateComInterfaceFlags
+    {
+        None = 0,
+
+        /// <summary>
+        /// The caller will provide an IUnknown Vtable.
+        /// </summary>
+        CallerDefinedIUnknown = 1,
+
+        /// <summary>
+        /// Flag used to indicate the COM interface should implement <see href="https://docs.microsoft.com/windows/win32/api/windows.ui.xaml.hosting.referencetracker/nn-windows-ui-xaml-hosting-referencetracker-ireferencetrackertarget">IReferenceTrackerTarget</see>.
+        /// When this flag is passed, the resulting COM interface will have an internal implementation of IUnknown
+        /// and as such none should be supplied by the caller.
+        /// </summary>
+        TrackerSupport = 2,
+    }
+
+    /// <summary>
+    /// Enumeration of flags for <see cref="ComWrappers.GetOrCreateObjectForComInstance(IntPtr, CreateObjectFlags)"/>.
+    /// </summary>
+    [Flags]
+    public enum CreateObjectFlags
+    {
+        None = 0,
+
+        /// <summary>
+        /// Indicate if the supplied external COM object implements the <see href="https://docs.microsoft.com/windows/win32/api/windows.ui.xaml.hosting.referencetracker/nn-windows-ui-xaml-hosting-referencetracker-ireferencetracker">IReferenceTracker</see>.
+        /// </summary>
+        TrackerObject = 1,
+
+        /// <summary>
+        /// Ignore the internal cache when creating an object.
+        /// </summary>
+        IgnoreCache = 2,
```

Comment:
Is this equivalent of `Marshal.GetUniqueObjectForIUnknown`? If yes, we should use "Unique" in this flags name - so that we use one name for the same concept.